### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by bmos**
In more in-depth refactoring I'm looking into, I have come across a number of changes needed to get tests passing, resolve test warnings, and obey lints. As requested in #1009, I have taken all the minor bugfixes and compiled them here.

[ddccd1e](https://github.com/move-coop/parsons/pull/1012/commits/ddccd1eb904375e39080f8ac7618364cf1318e1e) also fixes the only remaining issue holding back 3.11 support! So I added it to setup.py and the test suite to demonstrate that.

### CODE REVIEW

**Comment by shaunagm** [comment]
@bmos sorry for taking so long to respond, I've been traveling. Have you made any changes to the docs other than linting? It's hard to see in the diffs if there's anything other than line length changes.

**Review by shaunagm — APPROVED** [review]
LGTM, just waiting to confirm that there aren't any docs content changes before merging.

**Comment by bmos** [comment]
Only formatting changes and can revert those files if preferred.

**Comment by shaunagm** [comment]
No need to revert, just wanted to make sure I wasn't missing anything.

File: requirements.txt [hunk 1]
```diff
@@ -1,49 +1,48 @@
...
+censusgeocode==0.4.3.post1
 civis==1.16.0
-slackclient==1.3.0
-psycopg2-binary==2.9.3
-xmltodict==0.11.0
-gspread==3.7.0
-oauth2client==4.1.3
-google-auth==2.6.2
+dbt_redshift==1.4.0
+docutils<0.18,>=0.14
```
Comment (shaunagm) [inline requirements.txt:10]: @bmos do you remember why docutils is pinned like this? 

It's causing an issue bumping up sphinx in #1201 

https://app.circleci.com/pipelines/github/move-coop/parsons/6034/workflows/cba51139-5d87-4e55-9f68-ff53cf80afd9/jobs/7622?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link&utm_content=summary

Comment (bmos) [reply requirements.txt:10]: I think it's been that way since before my time

Comment (shaunagm) [reply requirements.txt:10]: git blame letting me down :( I will give it a try

**Pull request merged**